Koha/Plugin/Fi/KohaSuomi/DI/PatronController.pm

```diff
@@ -553,26 +554,59 @@ sub validate_credentials {
     }
 
     my $dbh = C4::Context->dbh;
-    unless (C4::Auth::checkpw_internal($dbh, $userid, $password)) {
-        return $c->render(
-            status => 401, 
-            openapi => { error => "Login failed." }
-        );
-    }
-
+    
     my $patron = Koha::Patrons->find({ userid => $userid });
-    $patron = Koha::Patrons->find({ cardnumber => $userid }) unless $patron;
-
-    if ($patron && $patron->lost) {
-        return $c->render( 
-            status => 403, 
-            openapi => { 
-                error => "Patron's card has been marked as 'lost'. Access forbidden." 
+    
+    #Try login if account is not locked
+    if ($patron and !$patron->account_locked) {
```

Comment:
I believe this can be simplified quite significantly to improve readability and avoid duplicated checks:
```
    if (!$patron || $patron->account_locked || !C4::Auth::checkpw_internal($dbh, $userid, $password)) {
        $patron->update({ login_attempts => $patron->login_attempts + 1 });
        return $c->render(
            status => 401, 
            openapi => { error => "Login failed." }
        );
    }        

    # Credentials valid and account not locked
        
    # Check for lost card and return 403 if so
        
    # Update lastseen and login_attempts

    # Return patron information

}
```

With this change most of the comments further on will be outdated.